src/Sylius/Component/Core/Exception/MissingChannelConfigurationException.php

```diff
@@ -13,10 +13,23 @@
 
 namespace Sylius\Component\Core\Exception;
 
+use Sylius\Component\Core\Model\ChannelInterface;
+use Sylius\Component\Core\Model\ProductVariantInterface;
+
 final class MissingChannelConfigurationException extends \RuntimeException
 {
     public function __construct(string $message, ?\Exception $previousException = null)
     {
         parent::__construct($message, 0, $previousException);
     }
+
+    public static function getMissingChannelProductVariantPriceMessage(ChannelInterface $channel, ProductVariantInterface $productVariant): string
+    {
+        $message = sprintf('Channel %s has no price defined for product variant', $channel->getName());
+        if ($productVariant->getName() !== null) {
+            return $message . sprintf(' %s (%s)', $productVariant->getName(), $productVariant->getCode());
+        }
+
+        return $message . sprintf(' with code %s', $productVariant->getCode());
```

Comment:
I noticed only now that the message is not very appropriate and it can be improved, which will also reduce the amount of code here.

```suggestion
        return new self(sprintf('Product variant %s has no price defined for channel %s', $productVariant->getDescriptor(), $channel->getName()))
```